include/CppUTestExt/MockActualCall.h

```diff
@@ -42,79 +42,118 @@ class MockActualCall
-    virtual MockActualCall& withUnsignedLongIntParameter(const SimpleString& name, unsigned long int value)=0;
-    virtual MockActualCall& withLongLongIntParameter(const SimpleString& name, cpputest_longlong value)=0;
-    virtual MockActualCall& withUnsignedLongLongIntParameter(const SimpleString& name, cpputest_ulonglong value)=0;
-    virtual MockActualCall& withDoubleParameter(const SimpleString& name, double value)=0;
-    virtual MockActualCall& withStringParameter(const SimpleString& name, const char* value)=0;
-    virtual MockActualCall& withPointerParameter(const SimpleString& name, void* value)=0;
-    virtual MockActualCall& withFunctionPointerParameter(const SimpleString& name, void (*value)())=0;
-    virtual MockActualCall& withConstPointerParameter(const SimpleString& name, const void* value)=0;
-    virtual MockActualCall& withMemoryBufferParameter(const SimpleString& name, const unsigned char* value, size_t size)=0;
-
-    virtual bool hasReturnValue()=0;
-    virtual MockNamedValue returnValue()=0;
-
-    virtual bool returnBoolValueOrDefault(bool default_value)=0;
-    virtual bool returnBoolValue()=0;
-
-    virtual int returnIntValueOrDefault(int default_value)=0;
-    virtual int returnIntValue()=0;
-
-    virtual unsigned long int returnUnsignedLongIntValue()=0;
-    virtual unsigned long int returnUnsignedLongIntValueOrDefault(unsigned long int default_value)=0;
-
-    virtual long int returnLongIntValue()=0;
-    virtual long int returnLongIntValueOrDefault(long int default_value)=0;
-
-    virtual cpputest_ulonglong returnUnsignedLongLongIntValue()=0;
-    virtual cpputest_ulonglong returnUnsignedLongLongIntValueOrDefault(cpputest_ulonglong default_value)=0;
-
-    virtual cpputest_longlong returnLongLongIntValue()=0;
-    virtual cpputest_longlong returnLongLongIntValueOrDefault(cpputest_longlong default_value)=0;
-
-    virtual unsigned int returnUnsignedIntValue()=0;
-    virtual unsigned int returnUnsignedIntValueOrDefault(unsigned int default_value)=0;
-
-    virtual const char * returnStringValueOrDefault(const char * default_value)=0;
-    virtual const char * returnStringValue()=0;
-
-    virtual double returnDoubleValue()=0;
-    virtual double returnDoubleValueOrDefault(double default_value)=0;
-
-    virtual void * returnPointerValue()=0;
-    virtual void * returnPointerValueOrDefault(void * default_value)=0;
-
-    virtual const void * returnConstPointerValue()=0;
-    virtual const void * returnConstPointerValueOrDefault(const void * default_value)=0;
-
-    virtual void (*returnFunctionPointerValue())()=0;
-    virtual void (*returnFunctionPointerValueOrDefault(void (*default_value)()))()=0;
-
-    virtual MockActualCall& onObject(const void* objectPtr)=0;
+    virtual MockActualCall& withName(const SimpleString& name) = 0;
+    virtual MockActualCall& withCallOrder(unsigned int callOrder) = 0;
+    MockActualCall& withParameter(const SimpleString& name, bool value)
```

Comment:
Ouch. Could we set AllowShortFunctionsOnASingleLine  to SFS_Inline so these don't get expanded?